composeApp/src/androidMain/kotlin/com/github/familyvault/services/QrCodeService.android.kt

```diff
@@ -30,8 +32,11 @@ class QrCodeService(private val context: Context) : IQRCodeService {
     override suspend fun scanPayload(): AddFamilyMemberDataPayload {
         val scannedResult = scanQRCode()
 
-        if (scannedResult.error != null || scannedResult.content == null) {
-            throw Exception(scannedResult.error)
+        if (scannedResult.status == QrCodeScanResponseStatus.CANCELED) {
+            throw CancellationException("Scan canceled")
+        }
+        if (scannedResult.content == null || scannedResult.status == QrCodeScanResponseStatus.ERROR) {
+            throw Exception(scannedResult.error ?: "Unknown error")
```

Comment:
Ja bym może zrobił tutaj wyjątki stricte dla kodów Qr, np. `QrCodeCancellationExceptio`, `QrCodeBadScanException` coś w tym rodzaju. 
Wtedy będzie wiadomo, że jest to związane z QR kodem w kodzie przechwytującym `try ... catch`